src/Handler/CurlMultiHandler.php

```diff
@@ -117,12 +117,15 @@ public function tick()
         // Step through the task queue which may add additional requests.
         P\queue()->run();
 
-        if ($this->active &&
-            curl_multi_select($this->_mh, $this->selectTimeout) === -1
-        ) {
-            // Perform a usleep if a select returns -1.
-            // See: https://bugs.php.net/bug.php?id=61141
-            usleep(250);
+        if ($this->active) {
+            $fd = curl_multi_select($this->_mh, $this->selectTimeout);
```

Comment:
I'd rather have this patch ensure that this block of code sleeps for at least 250 microseconds, instead of sleeping for an additional 250 microseconds.

Before, it'd sleep for 1.000 seconds in the case that curl_multi_select normally handled no activity (and didn't unexpectedly exit earlier). After this, it'd sleep 1.00025 seconds.

- If a request took exactly  1.00010 seconds, it'd sleep unnecessarily for an extra 150 microseconds

```php
            $select_start = microtime(true);
            $fd = curl_multi_select($this->_mh, $this->selectTimeout);
            $elapsed_milliseconds = (microtime(true) - $select_start) * 1000000;
            if ($fd === -1 || $fd === 0) {
                // Perform a usleep if a select returns -1.
                // See: https://bugs.php.net/bug.php?id=61141
                // Select will return 0 instead of -1 for newer libcurl versions since PHP 7.1.23 and 7.2.11
                // See: https://bugs.php.net/bug.php?id=76480
                $sleep_time = 250 - $elapsed_milliseconds;
                if ($sleep_time >= 1) {
                    usleep($sleep_time);
                }
            }
```